### PR TITLE
Add Funding Rate Scanner skill

### PR DESCRIPTION
**Pull request by dagangtj**
## Summary

This PR adds a Funding Rate Scanner skill that helps identify profitable long positions by scanning for negative funding rates across all Binance perpetual contracts.

## Features

- ✅ Scans 583+ perpetual contracts in real-time
- ✅ Zero dependencies (pure Node.js with native https module)
- ✅ No API key required (uses public Binance endpoints)
- ✅ Automatic daily earnings calculation (3 funding payments)
- ✅ Sorted by profitability
- ✅ Tested and working

## Test Results

Successfully scanned 583 contracts and found 117 negative funding rate opportunities:
- Highest: KITEUSDT at -1.6055% (4.82% daily)
- Second: HUMAUSDT at -0.9330% (2.80% daily)
- Third: STEEMUSDT at -0.2451% (0.74% daily)

## Usage

```bash
node skills/funding-rate-scanner/scan.js
```

## Files Added

- `SKILL.md` - Skill documentation
- `scan.js` - Main scanner (executable)
- `test.js` - Test script
- `package.json` - Package metadata

## Why This Matters

Negative funding rates mean short sellers pay long holders. This skill helps traders identify these profitable opportunities automatically, potentially earning passive income just by holding long positions.

### CODE REVIEW

File: skills/funding-rate-scanner/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## Output Format
+
+```
+🔥 Negative Funding Rate Opportunities
+
+Symbol: BTCUSDT
+Funding Rate: -0.05%
+Next Funding: 2h 30m
+Daily Earnings: 0.15% (3 payments)
+Recommendation: Open long position
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/SKILL.md]: Can you update the wording here to something like "Observation: Negative funding currently favors longs receiving funding, subject to market risk"?

File: skills/funding-rate-scanner/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Symbol: ETHUSDT
+Funding Rate: -0.03%
+Next Funding: 2h 30m
+Daily Earnings: 0.09% (3 payments)
+Recommendation: Open long position
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/SKILL.md]: Same for this one

File: skills/funding-rate-scanner/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Output Format
+
+```
+🔥 Negative Funding Rate Opportunities
+
+Symbol: BTCUSDT
+Funding Rate: -0.05%
+Next Funding: 2h 30m
+Daily Earnings: 0.15% (3 payments)
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/SKILL.md]: Also `Daily Earnings` to something like `Indicative funding per day at current rate`

File: skills/funding-rate-scanner/scripts/analyze.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  });
+});
+
+function fetch(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, res => {
+      let data = '';
+      res.on('data', chunk => data += chunk);
+      res.on('end', () => resolve({ json: () => JSON.parse(data) }));
+    }).on('error', reject);
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/scripts/analyze.js]: It would be good to add basic HTTP status / timeout / payload validation handling for robustness.

File: skills/funding-rate-scanner/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Triggers
+
+Use this skill when:
+- User asks to "scan funding rates"
+- User asks to "check Binance funding"
+- User asks to "find negative funding opportunities"
+- User asks about "funding rate arbitrage"
+
+## Safety
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/SKILL.md:69]: Please add a stronger disclaimer that this is informational only and not investment advice.

**Review by dimitrisn442 — CHANGES_REQUESTED** [review]

**Comment by dagangtj** [comment]
Hi @dimitrisn442, thank you for the thorough review! I've addressed all the requested changes:\n\n1. Updated the output wording from 'Daily Earnings' to 'Indicative funding per day at current rate'\n2. Replaced 'Recommendation: Open long position' with 'Observation: Negative funding currently favors longs receiving funding, subject to market risk'\n3. Added a stronger disclaimer section in SKILL.md\n4. Enhanced analyze.js with HTTP status checks, a 10s timeout, and payload validation for robustness\n\nPlease let me know if there's anything else you'd like adjusted.

File: skills/funding-rate-scanner/scripts/analyze.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+// Detailed analysis for a specific coin
+const https = require('https');
+
+const coin = process.argv[2];
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/scripts/analyze.js:5]: `process.argv[2]` is raw user input. When concatenated directly into a URL query string, characters like `&`, `=`, and `#` can alter the URL structure.

While exploitability is currently limited by the hardcoded domain, this is a bad pattern that normalises unsafe input handling in a shared, reference codebase. Anyone who forks this skill and loosens the domain constraint immediately has an SSRF vulnerability.

Can you please validate input against a regex before use?

File: skills/funding-rate-scanner/package.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    "binance",
+    "funding-rate",
+    "arbitrage",
+    "crypto",
+    "trading"
+  ],
+  "author": "first-butler",
+  "license": "MIT",
+  "engines": {
+    "node": ">=14.0.0"
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/package.json:20]: Node 14 is end-of-life and no longer receives security updates, bug fixes, or official support. This makes applications more vulnerable and increasingly incompatible with modern libraries, frameworks, and tooling, many of which now require newer supported Node versions.  For long-term stability, security, and ecosystem compatibility, apps should target an actively supported Node.js version instead.

Could you please use a minimum version of at least 22 that is officially supported until April 2027?

File: skills/funding-rate-scanner/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
+# Funding Rate Scanner
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/SKILL.md:1]: Can you please have a look on the contribution README guide and add the frontmatter https://github.com/binance/binance-skills-hub#contribution?

File: skills/funding-rate-scanner/scripts/analyze.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    throw new Error('Invalid premiumIndex response: missing lastFundingRate');
+  }
+  if (!Array.isArray(rates)) {
+    throw new Error('Invalid fundingRate response: expected array');
+  }
+  if (!stats || typeof stats.lastPrice !== 'string') {
+    throw new Error('Invalid 24hr ticker response: missing lastPrice');
+  }
+
+  const rate = parseFloat(current.lastFundingRate) * 100;
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/scripts/analyze.js:33]: Presenting figures as "annual" rate to any audience (human or AI agent) without immediately adjacent risk context is misleading financial information. The projection is hypothetical, not a forecast and it should be clear that this is not a financial advise.

Could you please make the warning clear and right next to the projection output?

File: skills/funding-rate-scanner/test.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
+/**
+ * Test script for Funding Rate Scanner
+ */
+
+const scanner = require('./scan.js');
+
+async function test() {
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/test.js:8]: Could you please mock the Binance API instead and test edge cases such as empty response, 429, malformed JSON etc which cannot be reliably triggered against a real API.

**Review by dimitrisn442 — CHANGES_REQUESTED** [review]
Thanks for resolving the previous comments @dagangtj. I left a couple of more.

File: skills/funding-rate-scanner/package.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    "scan": "node scan.js"
+  },
+  "keywords": [
+    "binance",
+    "funding-rate",
+    "arbitrage",
+    "crypto",
+    "trading"
+  ],
+  "author": "first-butler",
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/package.json:17]: Could you also please align the author with your username?

File: skills/funding-rate-scanner/package.json [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+  "license": "MIT",
```
Comment (dimitrisn442) [inline skills/funding-rate-scanner/package.json:18]: Could you also add the LICENSE.md file?

**Review by dimitrisn442 — CHANGES_REQUESTED** [review]

**Comment by dimitrisn442** [comment]
Hi @dagangtj, thanks again for the contribution.

I’m going to close this PR for now since it has been open for about a month, the requested review changes have not been addressed and there has been no further update or activity.

Feel free to reopen this or submit a new PR once the requested changes are applied. Thanks!
